src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -13,38 +13,170 @@
 using Akka.Configuration;
 using Akka.Hosting;
 
+#nullable enable
 namespace Akka.Cluster.Hosting
 {
+    public sealed class Role
+    {
+        public Role(string name, int? minimumNumberOfMembers = null)
+        {
+            Name = name;
+            MinimumNumberOfMembers = minimumNumberOfMembers;
+        }
+
+        /// <summary>
+        ///     The role name
+        /// </summary>
+        public string Name { get; }
+        
+        /// <summary>
+        ///     <para>
+        ///         Minimum required number of members before the leader changes member status
+        ///         of 'Joining' members to 'Up'. Typically used together with
+        ///         <see cref="Cluster.RegisterOnMemberUp"/> to defer some action, such as starting actors,
+        ///         until the cluster has reached a certain size.
+        ///     </para>
+        ///     <b>Default:</b> 1
+        /// </summary>
+        public int? MinimumNumberOfMembers { get; set; }
+    }
+    
     /// <summary>
-    /// The set of options for enabling Akka.Cluster support.
+    ///     The set of options for enabling Akka.Cluster support.
     /// </summary>
     public sealed class ClusterOptions
     {
         /// <summary>
-        /// The akka.cluster.roles values.
+        ///     The akka.cluster.roles and akka.cluster.role values.
         /// </summary>
-        public string[] Roles { get; set; }
+        public Role[]? Roles { get; set; }
```

Comment:
Don't like the `Role` type - adds more friction to the configuration surface in order to expose a feature barely anybody uses. Users can configure this in HOCON if they really want it - please remove it from the UI surface area entirely and go back to using simple strings.